### DISCUSSION TITLE
What is latest status of PPO mini-batch/batch support?

### DISCUSSION THREAD

**Discussion by profPlum** [tensorflow/agents#688] [Q&A] [answered]
More specifically if it doesn't support mini-batch learning then what is supported? Batch learning? Single observation learning?
Also if it is not yet implemented: is that just for ppo_clip_agent and/or ppo_kl_penalty_agent? Or does it also apply to base PPOAgent as well?

Context: I hoped to use TFAgents for a highly parallel project & part of that involved batch prediction & learning.
Also I saw a code comment in PPOAgent.py that says: "compute_value_and_advantage_in_train: [...] This argument must be set to `False` if mini batch learning is enabled." **Which suggests that mini-batch _might be supported_...?**

**Comment by sguada** [comment, marked as answer]
Yes mini-batch is supported, check https://github.com/tensorflow/agents/tree/master/tf_agents/examples/ppo

**Reply by profPlum** [reply]
That is great news! So just double checking: the 'disclaimer' for PPOClipAgent & PPOKLPenaltyAgent indicating otherwise 
was just outdated information? EDIT: I guess so since I cannot find them anymore... Thanks!
